### DISCUSSION TITLE
How to use GSI on batchGet?

### DISCUSSION THREAD

**Discussion by keneth-old-st** [sensedeep/dynamodb-onetable#525] [Q&A]
Hi everyone,

I am trying to create a batch get query that utilizes GSIs, but when I try to execute it, it gives me an error.

Below is a sample code on how I implemented it, once the `get` method has been called it will give me an error

The values on the GSIs are simply a concatenated representation of the fields used in the Model

```
const batch = {};
this.table.get({ GSI1PK: 'gsipkvaluehere', GSI1SK: 'gsiskvaluehere' }, {batch, index: 'GSI1', follow: true});
```
```
ERROR [ExceptionsHandler] Need complete keys for batched get operation
OneTableError: Need complete keys for batched get operation
```

does batchGet only works on the default PK and SK?

**Comment by mobsense** [comment]
Use find instead of get. With a GSI you can do a query/scan, but not get.
